### PR TITLE
SAK-45634 -User membership: Needs to constrain search size and UX imp…

### PR DESCRIPTION
**Pull request by mpellicer**
…rovements.

### CODE REVIEW

File: usermembership/tool/src/webapp/usermembership/userlist.jsp [hunk 1]
```diff
@@ -30,48 +31,40 @@
...
-					onkeypress="document.forms[0].submit;" value="#{msgs.bar_search}"
-					styleClass="active" style="margin-left: 3px !important;"
-					onclick="SPNR.disableControlsAndSpin( this, null );" />
-				<h:commandButton id="clearSearchButton" action="#{UserListBean.processActionClearSearch}"
-					onkeypress="document.forms[0].submit;" value="#{msgs.bar_clear_search}"
-					rendered="#{UserListBean.renderClearSearch}" style="margin-left: 3px !important;"
-					onclick="SPNR.disableControlsAndSpin( this, null );" />					
-	        </t:div>
-	 	</h:panelGrid>
-
```
Comment (fostersdesign) [inline usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Although updating the UI is a bit out of the scope of this issue, if you are fixing the layout of the table search and filters, please use the existing sakai-table-toolBar classes that take care of the layout and responsiveness for you. Worksite Setup's main page is a good example of this in use, which was added in #5865 based on the work in #5744. 

I can submit a separate Jira issue for this if you deem it to be out-of-scope for this PR.

Comment (mpellicer) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: I agree it's super out of the scope of the issue but I think that if I have to work on something anyways and I have the chance to upgrade an old UI for something better it's worth the effort, I have to work in the tool anyway to fix the issue and we don't see the same UI again and again, Sakai is super big so working on improving small chuncks everytime we visit them it's just fine.

I'm fine reverting all the UX stuff and just fixing the bug, I've splitted the PR in three separate commits indicating bugfix, JS improvement about the search input and the UX improvements.

I tried to use the work of #5865 and I think looks worse but maybe I messed up with something:
image

vs
image

https://github.com/mpellicer/sakai-apereo/commit/a901f026c0bfc11ef81b5aa92f559b57ccefc053

Comment (fostersdesign) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Okay. Stick with your original design and we can update it later. Thanks.

Comment (mpellicer) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: @fostersdesign I'm fine investing more time on this! do you see anything wrong in the separate branch applying the stules?

Comment (fostersdesign) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Sorry, where do I find that branch?

Comment (mpellicer) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: https://github.com/mpellicer/sakai-apereo/commit/a901f026c0bfc11ef81b5aa92f559b57ccefc053

Comment (fostersdesign) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Thanks. I'll take a look and let you know.

Comment (fostersdesign) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: You're really close. You just need to:
- remove the buttonContainer div (not needed if you don't have any buttons to put in it)
- move sakai-table-pagerContainer outside of sakai-table-filterContainer

Looks like we could add a whitespace: nowrap to the .sakai-table-toolBar .sakai-table-filterContainer label rule too:

`.sakai-table-toolBar .sakai-table-filterContainer label {
  white-space: nowrap;
}`

in _tables.scss (/library/src/morpheus-master/sass/modules/tool/_table.scss line 76)

Comment (mpellicer) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Thank you Shawn! Looks way better:

image

Comment (mpellicer) [reply usermembership/tool/src/webapp/usermembership/userlist.jsp:74]: Awwww deleted the branch by mistake, new PR here: https://github.com/sakaiproject/sakai/pull/9754

**Pull request closed without merging**